lib/ansible/plugins/action/package.py

```diff
@@ -38,31 +40,47 @@ def run(self, tmp=None, task_vars=None):
         self._supports_async = True
 
         result = super(ActionModule, self).run(tmp, task_vars)
-        del tmp  # tmp no longer has any effect
 
         module = self._task.args.get('use', 'auto')
 
-        if module == 'auto':
-            try:
-                if self._task.delegate_to:  # if we delegate, we should use delegated host's facts
-                    module = self._templar.template("{{hostvars['%s']['ansible_facts']['pkg_mgr']}}" % self._task.delegate_to)
-                else:
-                    module = self._templar.template('{{ansible_facts.pkg_mgr}}')
-            except Exception:
-                pass  # could not get it from template!
-
         try:
             if module == 'auto':
-                facts = self._execute_module(
-                    module_name='ansible.legacy.setup',
-                    module_args=dict(filter='ansible_pkg_mgr', gather_subset='!all'),
-                    task_vars=task_vars)
-                display.debug("Facts %s" % facts)
-                module = facts.get('ansible_facts', {}).get('ansible_pkg_mgr', 'auto')
-
-            if module != 'auto':
+
+                if self._task.delegate_to:
+                    # TODO: template delegate_to if needed
```

Comment:
We shouldn't need to worry with `delegate_to` not being a template, the TE should already be handling this as needed.